packages/logger/src/middleware/middy.ts

```diff
@@ -74,10 +68,11 @@ const injectLambdaContext = (
 
   const injectLambdaContextAfterOrOnError = async (): Promise<void> => {
     if (isClearState) {
-      loggers.forEach((logger: Logger, index: number) => {
+      loggers.forEach((logger: Logger) => {
+        logger.clearState();
         Logger.injectLambdaContextAfterOrOnError(
           logger,
-          persistentAttributes[index],
+          logger.getPersistentLogAttributes(),
```

Comment:
If I understand the logic here and in Logger correctly, this method injects `logEvent` and `clearState` parameters for the next invocations of the handler, not only to insert a snapshot of log attributes. In case of an error, will the `before` method executed? In `Logger` before function, it only injects `logEvent, but middy seems to inject both `logEvent` and `clearState` before.

Reply:
I went to check and the current implementation of the method on `main` looks like this:

```ts
public static injectLambdaContextAfterOrOnError(
    logger: Logger,
    initialPersistentAttributes: LogAttributes,
    options?: InjectLambdaContextOptions
  ): void {
    if (options && options.clearState === true) {
      logger.setPersistentLogAttributes(initialPersistentAttributes);
    }
  }
```

Given that you have added (correctly) the `logger.clearState()` right above this call (~L72), then calling the method is redundant as it would do the same thing.